roles/kafka_controller/tasks/main.yml

```diff
@@ -0,0 +1,447 @@
+  tags:
+    - log
+
+- name: Configure logredactor
+  include_role:
+    name: common
+    tasks_from: configure_logredactor.yml
+  vars:
+    log4j_file: "{{kafka_controller.log4j_file}}"
+    logredactor_rule_var: "{% if logredactor_rule_url == '' %}{{logredactor_rule_path}}{% else %}{{logredactor_rule_url}}{% endif %}"
+    logredactor_policy_refresh_interval_var: "{{logredactor_policy_refresh_interval}}"
+  with_items: "{{ kafka_controller_logredactor_logger_specs_list }}"
+  when: (kafka_controller_custom_log4j|bool) and (logredactor_enabled|bool)
+  tags:
+    - log
+
+- name: Restart kafka controller
+  ansible.builtin.debug:
+    msg: "restarting kafka controller"
+  notify: restart kafka
+  when: (kafka_controller_custom_log4j|bool) and (logredactor_enabled|bool) and (not kafka_controller_skip_restarts|bool)
+  tags:
+    - log
+
+- name: Create Kafka Controller Jolokia Config
+  template:
+    src: kafka_jolokia.properties.j2
+    dest: "{{kafka_controller_jolokia_config}}"
+    mode: 0640
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+  when: kafka_controller_jolokia_enabled|bool
+  notify: restart kafka
+  diff: "{{ not mask_sensitive_diff|bool }}"
+  tags:
+    - configuration
+
+- name: Create Kafka Controller Jaas Config
+  template:
+    src: kafka_server_jaas.conf.j2
+    dest: "{{kafka_controller.jaas_file}}"
+    mode: 0640
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+  when:
+    - "'GSSAPI' in kafka_controller_sasl_enabled_mechanisms"
+  notify: restart kafka
+  tags:
+    - configuration
+
+- name: Deploy JMX Exporter Config File
+  template:
+    src: "{{kafka_controller_jmxexporter_config_source_path}}"
+    dest: "{{kafka_controller_jmxexporter_config_path}}"
+    mode: 0640
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+  when: kafka_controller_jmxexporter_enabled|bool
+  tags:
+    - configuration
+
+- name: Create Service Override Directory
+  file:
+    path: "{{kafka_controller.systemd_override | dirname }}"
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+    state: directory
+    mode: 0640
+  tags:
+    - systemd
+
+- name: Write Service Overrides
+  template:
+    src: override.conf.j2
+    dest: "{{ kafka_controller.systemd_override }}"
+    mode: 0640
+    owner: root
+    group: root
+  notify: restart kafka
+  diff: "{{ not mask_sensitive_diff|bool }}"
+  tags:
+    - systemd
+
+- name: Create sysctl directory on Debian distributions
+  file:
+    path: /usr/lib/sysctl.d
+    state: directory
+  when: ansible_distribution == "Debian"
+  tags:
+    - sysctl
+    - privileged
+
+- name: Create sysctl directory on Debian distributions
+  file:
+    path: /usr/lib/sysctl.d
+    state: directory
+    mode: 0755
+  when: ansible_distribution == "Debian"
+  tags:
+    - sysctl
+    - privileged
+
+- name: Tune virtual memory settings
```

Comment:
let's get this dictionary `kafka_controller_sysctl` reviewed with the Kraft team.